unstructured_ingest/v2/processes/connectors/qdrant.py

```diff
@@ -0,0 +1,197 @@
+                **flatten_dict(
+                    data,
+                    separator="-",
+                    flatten_lists=True,
+                ),
+            },
+        }
+
+    def run(
+        self,
+        elements_filepath: Path,
+        file_data: FileData,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        with open(elements_filepath) as elements_file:
+            elements_contents = json.load(elements_file)
+
+        conformed_elements = [self.conform_dict(data=element) for element in elements_contents]
+        output_path = Path(output_dir) / Path(f"{output_filename}.json")
+
+        with open(output_path, "w") as output_file:
+            json.dump(conformed_elements, output_file)
+        return output_path
+
+
+class QdrantUploaderConfig(UploaderConfig):
+    batch_size: int = Field(default=50, description="Number of records per batch")
+    num_processes: Optional[int] = Field(
+        default=1,
+        description="Optional limit on number of threads to use for upload",
+        deprecated=True,
+    )
+
+
+@dataclass
+class QdrantUploader(Uploader):
+    connector_type: str = CONNECTOR_TYPE
+    upload_config: QdrantUploaderConfig
+    connection_config: QdrantConnectionConfig
+
+    @DestinationConnectionError.wrap
+    def precheck(self) -> None:
+        async def check_connection():
+            async_client = self.connection_config.get_async_client()
+            await async_client.get_collections()
+
+        asyncio.run(check_connection())
+
+    def is_async(self):
+        return True
+
+    async def run_async(
+        self,
+        path: Path,
+        file_data: FileData,
+        **kwargs: Any,
+    ) -> Path:
+        with path.open("r") as file:
+            elements: list[dict] = json.load(file)
+
+        logger.debug("Loaded %i elements from %s", len(elements), path)
+
+        batches = list(batch_generator(elements, batch_size=self.upload_config.batch_size))
+        logger.debug(
+            "Elements split into %i batches of size %i.",
+            len(batches),
+            self.upload_config.batch_size,
+        )
+        await asyncio.gather(*[self._upsert_batch(batch) for batch in batches])
+
+    async def _upsert_batch(self, batch: list[dict]) -> None:
+        from qdrant_client import models
+
+        client = self.connection_config.get_async_client()
+        points: list[models.PointStruct] = [models.PointStruct(**item) for item in batch]
+        try:
+            logger.debug(
+                "Upserting %i points to the '%s' collection.",
+                len(points),
+                self.connection_config.collection_name,
+            )
+            response = await client.upsert(
+                self.connection_config.collection_name, points=points, wait=True
+            )
+        except Exception as api_error:
+            logger.debug("Upsert response: %s", response)
```

Comment:
If this errors, `response` might not exist.

Reply:
True, I'll drop that from here